hazelcast/src/main/java/com/hazelcast/cache/impl/AbstractCacheProxyBase.java

```diff
@@ -141,9 +141,6 @@ public void close() {
         }
         loadAllTasks.clear();
```

Comment:
Hmm, I don't see a problem here; we just wait for each `Future` in `loadAllTasks` to complete and then clear the list. The list itself is a `CopyOnWriteArrayList` (so thread-safe) and moreover, in https://github.com/vbekiaris/hazelcast/blob/db5ea7d2fe437fcac8ad1304aeca0925ba8471ee/hazelcast/src/main/java/com/hazelcast/cache/impl/AbstractCacheProxyBase.java#L128-L128 we mark the cache as closed so no further tasks can be submitted.